conda_forge_tick/update_upstream_versions.py

```diff
@@ -113,35 +113,50 @@ class RawURL:
     name = 'RawURL'
 
     def get_url(self, meta_yaml):
+        if 'feedstock_name' not in meta_yaml:
+            return None
+        if 'version' not in meta_yaml:
+            return None
         pkg = meta_yaml['feedstock_name']
         url_template = "https://raw.githubusercontent.com/conda-forge/{}-feedstock/master/recipe/meta.yaml"
         try:
-            content = requests.get(url_template.format(pkg)).content
+            resp = requests.get(url_template.format(pkg))
+            resp.raise_for_status()
+            content = resp.content
             content = content.decode('utf-8')
         except Exception:
             return None
 
         current_ver = meta_yaml['version']
         orig_ver = current_ver
         found = True
-        while found:
+        count = 0
+        max_count = 10
+        while found and count < max_count:
```

Comment:
Could you explain this loop? I'm not certain I understand what is going on.

Reply:
Some servers don't return error codes for HEAD requests and therefore the algorithm will think the versions exist. For eg: it will find 1.1.11, 1.1.12,1.13, and so on going into an infinite loop.